orchagent/portsorch.cpp

```diff
@@ -4758,7 +4779,7 @@ bool PortsOrch::initializePort(Port &port)
     }
 
     /* initialize port admin speed */
-    if (!getPortSpeed(port.m_port_id, port.m_speed))
+    if (!isAutoNegEnabled(port.m_port_id) && !getPortSpeed(port.m_port_id, port.m_speed))
```

Comment:
I don't understand the intention here. Are we getting speed only inf autoneg is not enabled? The title says get adv speed but I don't see get adv speed not called. Can you please explain more?

Reply:
We are polling for `SAI_PORT_ATTR_SPEED` speed only when auto neg is not enabled. 

https://github.com/opencomputeproject/SAI/blob/566d4a8ef2e90bee986133042c4f8c2d7478a107/inc/saiport.h#L772

As when autoneg is enabled, `SAI_PORT_ATTR_SPEED` is not set. We set and poll `SAI_PORT_ATTR_ADVERTISED_SPEED`

https://github.com/opencomputeproject/SAI/blob/566d4a8ef2e90bee986133042c4f8c2d7478a107/inc/saiport.h#L819

As for the GET call on adv speed, that is done here:
https://github.com/sonic-net/sonic-swss/blob/e04bb436eabe135eca2482852dc492599432d44b/orchagent/portsorch.cpp#L3464